documentation/source/simulator_support.rst

```diff
@@ -131,6 +131,37 @@ For Verilator 4.102 and above, the `-CFLAGS -DVM_TRACE_FST=1` argument is no lon
 
 The resulting file will be ``dump.fst`` and can be opened by ``gtkwave dump.fst``.
 
+.. _sim-verilator-user-clocks:
+
+C++ clocks
+----------
+
+For increased performance, clocks may be generated in C++ with a user-defined callback function.
+First define a callback matching the cocotb-provided declaration:
+
+  .. code-block:: cpp
+
+    #include <memory>       // unique_ptr
+    #include "Vtop.h"
+    #include "verilated.h"  // vluint64_t
+
+    vluint64_t user_clock_cb(std::unique_ptr<Vtop> & topp, vluint64_t current_time) {
+        if (current_time % CLK_HALFPERIOD == 0) {
+            topp->clk = !topp->clk;
+        }
```

Comment:
Do you need this protection here since the main loop only calls the user_clock_cb on the next clock edge by design?
```suggestion
        topp->clk = !topp->clk;
```

Reply:
Not in this single-clock example.

Reply:
Perhaps the example should be expanded for multiple clocks and skew? Something that users can copy-paste-modify for their own tests.